### ISSUE TITLE
Allow objects to be passed in as an inputs for network.train

### ISSUE THREAD

**Issue by AkashSamlal** [liquidcarrot/carrot#129]
**Allow objects to be passed into an argument for inputs and outputs for network.train**

**Comment by nicszerman** [comment]
I'm trying to understand your idea. How would the neural network process the input obiect?

**Comment by AkashSamlal** [comment]
> I'm trying to understand your idea. How would the neural network process the input object?

As of currently, we can only pass in a training set as only arrays like this: 
![image](https://user-images.githubusercontent.com/43329669/60399931-b34bab80-9b3a-11e9-997b-96480d34abdf.png)

However, we should also make it a way for objects to be treated the same, for instance:
![image](https://user-images.githubusercontent.com/43329669/60399948-0160af00-9b3b-11e9-9ed7-42f78e8a5638.png)

**Comment by nicszerman** [comment]
Makes sense! I like your idea. We need to think if it's best to add this
functionality under .train() or if we should add it under another layer of
abstraction.

*Nicholas Szerman*

**Comment by tracycollins** [comment]
This relates to my request to add labels/names to inputs and outputs

**Comment by christianechevarria** [comment]
Related to #85 

Specifically, @tracycollins comment:

> Also, an unrelated item that would be nice to have: labels for input and output nodes. I think they are only identified by array index right now? It would help with testing various networks in parallel that have different inputs.
> 
> Right now, I add my own wrapper to the network object that provides this feature (among other things).

**Comment by christianechevarria** [comment]
@tracycollins asking about the wrapper you wrote, is there any feature you incorporated that you would want to see implemented in the library?

**Comment by tracycollins** [comment]
@christianechevarria The only feature that I found that I really used was the labelling of the inputs/outputs. Thanks!
